### PR TITLE
Add more mandatory field and add validation

### PR DESCRIPTION
**Pull request by sndrgrdn**

### CODE REVIEW

**Review by baukevw — APPROVED** [review]
Nice work! But sad that you have to do this...

File: spec/resources/invoices/invoices_api_spec.rb [hunk 1]
```diff
@@ -12,7 +12,15 @@
   end
 
   it "is valid with certain attributes" do
-    sales_invoice = Elmas::SalesInvoice.new(journal:"my-awesome-journal", ordered_by: "1230")
+    sales_invoice = Elmas::SalesInvoice.new(
+      journal:"my-awesome-journal",
+      ordered_by: "1230",
+      invoice_date: Time.now,
```
Comment (loed-idzinga) [inline spec/resources/invoices/invoices_api_spec.rb:18]: I know this is just filling some fields to be present, but try to use `zone` whenever you use `Time`.

```suggestion
      invoice_date: Time.zone.now,
```

File: spec/resources/invoices/invoices_api_spec.rb [hunk 2]
```diff
@@ -12,7 +12,15 @@
...
+      order_date: Time.now,
```
Comment (loed-idzinga) [inline spec/resources/invoices/invoices_api_spec.rb:19]:
```suggestion
      order_date: Time.zone.now,
```

File: spec/resources/bank_accounts/bank_accounts_api_spec.rb [hunk 3]
```diff
@@ -41,14 +41,14 @@
     end
 
     it "should apply given filters for find_by" do
-      expect(Elmas).to receive(:get).with("crm/BankAccounts?$filter=Account+eq+guid'12abcdef-1234-1234-1234-123456abcdef'&$filter=ID+eq+guid'12abcdef-1234-1234-1234-123456abcdef'")
+      expect(Elmas).to receive(:get).with("crm/BankAccounts?$filter=Account eq guid'12abcdef-1234-1234-1234-123456abcdef'&$filter=ID eq guid'12abcdef-1234-1234-1234-123456abcdef'")
```
Comment (loed-idzinga) [inline spec/resources/bank_accounts/bank_accounts_api_spec.rb:44]: Are these URL's properly coded? I think it should be something like `%20` or am I missing something?

*This comment applies to all the URL changes*

File: spec/resources/bank_entry_lines/bank_entry_lines_spec.rb [hunk 4]
```diff
@@ -24,7 +24,15 @@
...
   it "is valid with mandatory attributes" do
-    bank_entry_line = Elmas::BankEntryLine.new(amount_FC: "23", entry_ID: "23299ask-2233", GL_account: "sdjkj29")
+    bank_entry_line = Elmas::BankEntryLine.new(
+      amount_FC: "23",
+      entry_ID: "23299ask-2233",
+      GL_account: "sdjkj29",
+      account: "123",
+      description: "foobar",
+      our_ref: "123",
+      date: Time.now
```
Comment (loed-idzinga) [inline spec/resources/bank_entry_lines/bank_entry_lines_spec.rb:34]:
```suggestion
      date: Time.zone.now
```

**Review by loed-idzinga — COMMENTED** [review]
Nice work on updating the gem! 💪 Some feedback:

File: spec/resources/invoices/invoices_api_spec.rb (hunk 1 above)
Comment (sndrgrdn) [reply spec/resources/invoices/invoices_api_spec.rb:18]: We are not in Rails land! `zone` comes from activesupport

File: spec/resources/bank_accounts/bank_accounts_api_spec.rb (hunk 3 above)
Comment (sndrgrdn) [reply spec/resources/bank_accounts/bank_accounts_api_spec.rb:44]: Screenshot 2022-09-01 at 10 04 18

Comment (loed-idzinga) [reply spec/resources/bank_accounts/bank_accounts_api_spec.rb:44]: Oh .. I did not say anything 😂

**Review by loed-idzinga — APPROVED** [review]
Awesome work! 👍

**Pull request merged**